### PR TITLE
Pip: update error types and messages

### PR DESCRIPTION
**Pull request by chmeliik**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] New code has type annotations
- [ ] ~Docs updated (if applicable)~
- [ ] ~Docs links in the code are still valid (if docs were updated)~
- [ ] ~[Issue](https://github.com/containerbuildsystem/cachi2/issues/13) that tracks breaking changes to the OSBS2 integration is updated (if applicable)~

### CODE REVIEW

File: cachi2/core/utils.py [hunk 1]
```diff
@@ -6,12 +6,15 @@
...
 from typing import Iterator, Union
 
-from cachi2._compat.errors import UnknownHashAlgorithm
 from cachi2.core.config import get_worker_config
 
 log = logging.getLogger(__name__)
 
 
+class UnknownHashAlgorithm(ValueError):
+    """The hash algorithm is unknown by Cachi2."""
```
Comment (chmeliik) [inline cachi2/core/utils.py:15]: This is just meant to be a "more specific ValueError", not a user-facing error type. OK to keep it in this file?

Comment (lkolacek) [reply cachi2/core/utils.py:15]: I would probably place it to `core/errors.py` to others but I am fine keeping it here. :)

Comment (chmeliik) [reply cachi2/core/utils.py]: Solved this differently. The `utils.hash_file` saves 3 lines at best, removed it instead

Comment (chmeliik) [reply cachi2/core/utils.py]: (And that removes the need for an UnknownHashAlgorithm error)

**Review by taylormadore — APPROVED** [review]
LGTM :+1:

**Pull request merged**